nimble/host/include/host/ble_l2cap.h

```diff
@@ -234,33 +470,166 @@ struct ble_l2cap_event {
+ * When a connection request is received, the provided callback function will be invoked with the corresponding
+ * event information.
+ *
+ * @param psm       The Protocol/Service Multiplexer (PSM) for the server.
+ * @param mtu       The Maximum Transmission Unit (MTU) size for the server.
+ * @param cb        Pointer to the callback function to be invoked when a connection request is received.
+ * @param cb_arg    An optional argument to be passed to the callback function.
+ *
+ * @return          0 on success;
+ *                  A non-zero value on failure.
+ */
 int ble_l2cap_create_server(uint16_t psm, uint16_t mtu,
                             ble_l2cap_event_fn *cb, void *cb_arg);
 
+/**
+ * @brief Initiate an L2CAP connection.
+ *
+ * This function initiates an L2CAP connection to a remote device with the specified connection handle,
+ * Protocol/Service Multiplexer (PSM), Maximum Transmission Unit (MTU) size, and receive SDU buffer.
+ * When the connection is established or if there is an error during the connection process, the provided
+ * callback function will be invoked with the corresponding event information.
+ *
+ * @param conn_handle   The connection handle for the remote device.
+ * @param psm           The Protocol/Service Multiplexer (PSM) for the connection.
+ * @param mtu           The Maximum Transmission Unit (MTU) size for the connection.
+ * @param sdu_rx        Pointer to the receive Service Data Unit (SDU) buffer.
+ * @param cb            Pointer to the callback function to be invoked when the connection is established.
+ * @param cb_arg        An optional argument to be passed to the callback function.
+ *
+ * @return              0 on success;
+ *                      A non-zero value on failure.
+ */
 int ble_l2cap_connect(uint16_t conn_handle, uint16_t psm, uint16_t mtu,
                       struct os_mbuf *sdu_rx,
                       ble_l2cap_event_fn *cb, void *cb_arg);
+
+/**
+ * @brief Disconnect an L2CAP channel.
+ *
+ * This function disconnects the specified L2CAP channel by sending a disconnect signal.
+ *
+ * @param chan          Pointer to the L2CAP channel structure representing the channel to disconnect.
+ *
+ * @return              0 on success;
+ *                      A non-zero value on failure.
+ */
 int ble_l2cap_disconnect(struct ble_l2cap_chan *chan);
+
+/**
+ * @brief Send an SDU (Service Data Unit) over an L2CAP channel.
+ *
+ * This function sends an SDU over the specified L2CAP channel. The SDU is encapsulated
+ * in L2CAP frames and transmitted to the remote device.
+ *
+ * @param chan          Pointer to the L2CAP channel structure representing the channel to send the SDU on.
+ * @param sdu_tx        Pointer to the os_mbuf structure containing the SDU (Service Data Unit) to send.
+ *
+ * @return              0 on success;
+ *                      BLE_HS_ESTALLED: if the transmit buffer contained an SDU that has not been entirely sent,
```

Comment:
this is not correct,    host will return ESTALLED when there was not enough credits available to send whole SDU,  in such case application needs to wait for BLE_L2CAP_EVENT_COC_TX_UNSTALLED event before being able to transmit more data


(and BLE_L2CAP_COC_FLAG_STALLED is implementation internal detail)